helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -70,7 +70,7 @@ protected AssignmentMetadataStore(BucketDataAccessor bucketDataAccessor, String
         _globalBaseline = splitAssignments(baseline);
       } catch (ZkNoNodeException ex) {
         // Metadata does not exist, so return an empty map
-        _globalBaseline = Collections.emptyMap();
+        _globalBaseline = new HashMap<>();
```

Comment:
This change is non-essential, right?

Reply:
It is. In this PR, I changed the logic so the map will be kept. It needs to be modifiable.
Another option is to introduce more logic to handle the corner cases, which I don't prefer.